### PR TITLE
Improve references in mutual recursion

### PR DESCRIPTION
**Pull request by Janpot**
Fixes https://github.com/StefanTerdell/zod-to-json-schema/issues/57

This is a second take on https://github.com/StefanTerdell/zod-to-json-schema/pull/59

### CODE REVIEW

**Comment by Janpot** [comment]
@StefanTerdell Looks like the 10.x build broke in https://github.com/StefanTerdell/zod-to-json-schema/commit/57dcca17eec5120ce407630284e803efaf2d2f73

File: src/parseDef.ts [hunk 1]
```diff
@@ -72,11 +72,12 @@ export type JsonSchema7Type = JsonSchema7TypeUnion & JsonSchema7Meta;
 
 export function parseDef(
   def: ZodTypeDef,
-  refs: Refs
+  refs: Refs,
+  isTop?: boolean
 ): JsonSchema7Type | undefined {
   const seenItem = refs.seen.find((x) => Object.is(x.def, def));
```
Comment (Janpot) [inline src/parseDef.ts:78]: Just something I noticed. Not sure you've been doing any performance benchmarks, but I've seen this pattern not scale well before. (`.find`/`.includes`/`.indexOf`/..., being called on a growing array in a nested loop).

Comment (StefanTerdell) [reply src/parseDef.ts:78]: I guess we could just switch to a map

**Comment by StefanTerdell** [comment]
> @StefanTerdell Looks like the 10.x build broke in https://github.com/StefanTerdell/zod-to-json-schema/commit/57dcca17eec5120ce407630284e803efaf2d2f73

Also fixed in #62

**Comment by Janpot** [comment]
> Also fixed in https://github.com/StefanTerdell/zod-to-json-schema/pull/62

👍  I merged in the changes

**Review by StefanTerdell — CHANGES_REQUESTED** [review]

**Comment by StefanTerdell** [comment]
Hey @Janpot , being a complete bozo I forgot to commit the review so you adressed my loose comment but not the one about commenting in the main API file, explicitly setting the argument and/or changing the parameter name. Tagging you in case you missed it. Also feel free to update the contributors field of the package file (or I will do it after merging unless you ask me not to).

**Comment by Janpot** [comment]
No worries, I currently don't have more bandwidth left to go dig deeply into when and how `jsonSchema` gets initialized. I just copied this from another part of the codebase and it worked. (By which I mean it passed the tests). For the time being I can continue using my fork so it's not urgent, maybe in a few weekends I'll have time to pick this up again.

**Pull request merged**